kani-compiler/src/kani_middle/provide.rs

```diff
@@ -6,10 +6,10 @@
 
 use crate::kani_middle::reachability::{collect_reachable_items, filter_crate_items};
 use crate::kani_middle::stubbing;
+use crate::kani_middle::ty::query::query_provided::collect_and_partition_mono_items;
```

Comment:
This change looks a bit odd. Isn't this importing `collect_and_partition_mono_items` from `rustc_middle`? If so, why is it referring to `crate::kani_middle`?